Argus-Labs/world-engine
# refactor(cardinal): combine entry and entity into a single concept

## What is the purpose of the change

The current code distinguished between an "Entity" which uniquely identifies some object in the world and an "Entry" which combines an Entity with a "Location" (so data about the entity/entry can be saved and recovered). Entities also consist of an ID as well as a version number.

These different pieces can get confusing. This PR simplifies our object management by getting rid of the concept of "Entries" entirely. Now, an object in the world can be uniquely identified with a EntityID. An entity consists of an EntityID and a Location. 

In addition, all pointers to Entities or Location have been replaced with concrete objects. An Entity is small (consisting of 3 ints) and they should never be directly modified. EntityIDs never change and Location changes should be handled by the world storage layer.

The concept of versions as part of an ID has also been removed (it was not used in the code).

## Brief Changelog

- Commit 1: Replace Entity with and Entity ID. Remove Entry pointers in parameters and return values.
- Commit 2: Replace all instances of "entry" with "entity". Move the contents of entry.go into entity.go.

## Testing and Verifying

- View the first commit in isolation first. There are some logic changes as we no longer use pointers to Entry/Entities. Note, the Entry still exists in this commit.
- Verify the second commit just renames instances of Entry with Entity.

## Documentation and Release Note

- Does this pull request introduce a new feature or user-facing behavior changes? (no)
- Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
- How is the feature or change documented? (not documented)
